protos/flyteidl/admin/launch_plan.proto

```diff
@@ -52,63 +53,73 @@ message LaunchPlanList {
 
     // In the case of multiple pages of results, the server-provided token can be used to fetch the next page
     // in a query. If there are no more results, this value will be empty.
-    string token                     = 2;
+    string token = 2;
 }
 
 // Defines permissions associated with executions created by this launch plan spec.
 message Auth {
-  oneof method {
-    string assumable_iam_role         = 1;
-    string kubernetes_service_account = 2;
-  }
+    oneof method {
+        string assumable_iam_role = 1;
+        string kubernetes_service_account = 2;
+    }
 }
 
+// Defines allowed overrides for the workflow template.
+message WorkflowTemplateOverrides {
+    // If a field is set in metadata, it'll override the one from the original workflow template.
+    // If it's not-set, the system will continue to use the value from the original workflow template.
+    // TODO: Should we just require a full replacement of metadata to avoid having to deal with merges?
```

Comment:
This seems like something that should be decided before moving ahead with the changes. It might be hard to switch from merge to wholesale replacement once records exist using the partial data format.
Thoughts: 
* Merge is easier on the user
* Wholesale replacement records would be rendered incomplete when we add new fields to the metadata. I know protobuf has an 'all things are optional' approach, but practically speaking that's not always true. We would need to consider what it means for a new metadata field to be added and any existing launch plans to, by convention, exclude the value because they are replacing the entire metadata object and don't specify the value.

Reply:
@kumare3 @wild-endeavor thoughts?

Reply:
With respect to the specifica question at hand, I'd prefer to do a merge.  It's just slightly trickier, it's not exactly rocket science, and can be very fully unit tested.  Not worried about this at all.

What is the goal of this by the way?  Is this the start of a general solution for execution overrides and will one day include information on task level cpu/memory overrides, concurrency overrides, etc.?  What this PR has makes sense for the run to completion flag, but I haven't thought through how task level overrides will work in the future, and whether this approach will mesh with those needs seamlessly.

Reply:
This is an interesting question.  So @bnsblue just brought up and some users have, about the ability to override node metadata. This PR allows node metadata.
I know I added this as a comment, but lets hold off on this change. I think this deserves a small spec and I think @bnsblue has started recording some thoughts.

WDYT?

Reply:
also my two cents, merging is not a simple endeavor, replacement has the con that Randy mentioned, of overriding fields. But if the older spec does not consist it, newer one should override any non existent fields too right?

Reply:
I am linking the one-pager doc for your reference. I just started it yesterday so it is still pretty brief at the moment: https://docs.google.com/document/d/1bA9T_DHSE2_752N5Y-K8AtO4VZQ69aJDzGn274SkC6I/edit#